fabtests/common/shared.c

```diff
@@ -396,28 +397,59 @@ int ft_reg_mr(struct fi_info *fi, void *buf, size_t size, uint64_t access,
 	attr.offset = 0;
 	attr.requested_key = key;
 	attr.context = NULL;
-	attr.iface = opts.iface;
+	attr.iface = iface;
 
-	switch (opts.iface) {
+	switch (iface) {
 	case FI_HMEM_NEURON:
-		attr.device.neuron = opts.device;
+		attr.device.neuron = device;
 		break;
 	case FI_HMEM_ZE:
-		attr.device.ze = opts.device;
+		attr.device.ze = device;
 		break;
 	default:
 		break;
 	}
 
-	flags = (opts.iface) ? FI_HMEM_DEVICE_ONLY : 0;
+	flags = (iface) ? FI_HMEM_DEVICE_ONLY : 0;
 	ret = fi_mr_regattr(domain, &attr, flags, mr);
-	if (ret)
+	if (ret) {
+		FT_ERR("fi_mr_regattr failed: %d", ret);
 		return ret;
+	}
+
+	/* TODO: ft_alloc_msgs() is called before endpoint is allocated. Thus,
+	 * skip binding if ep is null.
+	 */
+	if (fi->domain_attr->mr_mode & FI_MR_ENDPOINT && ep) {
```

Comment:
I may need this commit https://github.com/ofiwg/libfabric/pull/7938/commits/0aafff5103a8cc3970d7dfe838f3156e78d926b8

Reply:
fabtest init calls ft_mr_reg for internal messaging buffers before the endpoint is allocated. Commit https://github.com/ofiwg/libfabric/commit/0aafff5103a8cc3970d7dfe838f3156e78d926b8 resovles this issue.